### PR TITLE
fix: deprecate ne and nin operators for query filters

### PR DESCRIPTION
**Pull request by khushal87**
## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] Code changes are tested

## Changelog

- The goal of the PR is to deprecate the `$ne` and the `$nin` operators for the query filters.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1423,7 +1423,7 @@ export type ReactionFilters<StreamChatGenerics extends ExtendableGenerics = Defa
 export type ChannelFilters<StreamChatGenerics extends ExtendableGenerics = DefaultGenerics> = QueryFilters<
   ContainsOperator<StreamChatGenerics['channelType']> & {
     members?:
-      | RequireOnlyOne<Pick<QueryFilter<string>, '$in' | '$nin'>>
+      | RequireOnlyOne<Pick<QueryFilter<string>, '$in'>>
```
Comment (szuperaz) [inline src/types.ts:1426]:
```suggestion
      | RequireOnlyOne<Pick<QueryFilter<string>, '$in' | '$nin'>>
```

Comment (szuperaz) [reply src/types.ts:1426]: Removing `$nin` here will technically make this a breaking change

Screenshot 2024-09-13 at 14 38 13

Comment (khushal87) [reply src/types.ts:1426]: In that case, I will keep it here for now

Comment (szuperaz) [reply src/types.ts]: Thanks

**Review by szuperaz — APPROVED** [review]

File: src/types.ts [hunk 2]
```diff
@@ -1649,14 +1649,26 @@ export type QueryFilter<ObjectType = string> = NonNullable<ObjectType> extends s
       $in?: PrimitiveFilter<ObjectType>[];
       $lt?: PrimitiveFilter<ObjectType>;
       $lte?: PrimitiveFilter<ObjectType>;
+      /**
+       * @deprecated handle the filter explicitly in the code. This is done for performance reasons.
+       */
       $ne?: PrimitiveFilter<ObjectType>;
+      /**
+       * @deprecated handle the filter explicitly in the code. This is done for performance reasons.
```
Comment (oliverlaz) [inline src/types.ts:1657]: Can you rephrase this line to something that gives more clarity?

`deprecated and will be removed in a future release. Filtering shall be applied client-side.`

Comment (khushal87) [reply src/types.ts]: Done

**Review by oliverlaz — APPROVED** [review]
